lib/howl/commands/app_commands.moon

```diff
@@ -665,39 +665,107 @@ do_search = (search, whole_word) ->
       m = matches[i]
       file_search_hit_to_location(m, search, display_as)
 
-  locations, searcher, project
+  locations, searcher
 
-command.register
-  name: 'project-file-search',
-  description: 'Searches files in the the current project'
-  input: (opts) ->
-    editor = app.editor
-    search = nil
-    whole_word = false
+ask_for_search_directory = (search_query) ->
+  buffer = app.editor.buffer
+  file = buffer.file or buffer.directory
+  project = Project.for_file(file) if file
 
-    if opts.text and not opts.text.is_empty
-      search = opts.text
+  opts =
+    title: if search_query
+      "(Please specify a directory to search for '#{search_query}'): "
+    else
+      "(Please specify a directory to search): "
+    prompt: "Directory: "
 
-    unless search or app.window.command_panel.is_active
-      if editor.selection.empty
-        search = app.editor.current_context.word.text
-        whole_word = true unless search.is_empty
-      else
-        search = editor.selection.text
+  start_dir = if project
+    project.root
```

Comment:
Follow up on top level [comment](https://github.com/howl-editor/howl/pull/559#issuecomment-603656120):

I think we should always use get_buffer_dir() as the start_dir for any search. Project dir searches already have their own commands. Yes, they wont let you pick the directory, but when picking a dir for folder-search, I'd prefer to start with the current dir and go up the tree if necessary, which is easy by just pressing `backspace`. WDYT?

Reply:
This is the behaviour that I'd started with but soon noticed that most of the time I navigated back to the project root anyway. I usually search 2-3 folders in the project root.

I've added a configuration variable to control this: folder_search_start_dir, which defaults to the directory of the current buffer.